server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistSnapshotsStep.java

```diff
@@ -0,0 +1,174 @@
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.step;
+
+import javax.annotation.Nullable;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.api.resources.Scopes;
+import org.sonar.api.utils.System2;
+import org.sonar.core.component.SnapshotDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.server.computation.batch.BatchReportReader;
+import org.sonar.server.computation.component.Component;
+import org.sonar.server.computation.component.DbIdsRepository;
+import org.sonar.server.computation.component.TreeRootHolder;
+import org.sonar.server.computation.period.Period;
+import org.sonar.server.computation.period.PeriodsHolder;
+import org.sonar.server.db.DbClient;
+
+/**
+ * Persist snapshots
+ * Also feed the components cache {@link DbIdsRepository} with snapshot ids
+ */
+public class PersistSnapshotsStep implements ComputationStep {
+
+  private final System2 system2;
+  private final DbClient dbClient;
+  private final TreeRootHolder treeRootHolder;
+  private final BatchReportReader reportReader;
+  private final DbIdsRepository dbIdsRepository;
+  private final PeriodsHolder periodsHolder;
+
+  public PersistSnapshotsStep(System2 system2, DbClient dbClient, TreeRootHolder treeRootHolder, BatchReportReader reportReader, DbIdsRepository dbIdsRepository,
+    PeriodsHolder periodsHolder) {
+    this.system2 = system2;
+    this.dbClient = dbClient;
+    this.treeRootHolder = treeRootHolder;
+    this.reportReader = reportReader;
+    this.dbIdsRepository = dbIdsRepository;
+    this.periodsHolder = periodsHolder;
+  }
+
+  @Override
+  public void execute() {
+    DbSession session = dbClient.openSession(false);
+    try {
+      Component root = treeRootHolder.getRoot();
+      ProcessPersistSnapshot processPersistSnapshot = new ProcessPersistSnapshot(session, reportReader, reportReader.readMetadata().getAnalysisDate());
+      processPersistSnapshot.process(root, null);
+      session.commit();
+    } finally {
+      session.close();
+    }
+  }
+
+  private class ProcessPersistSnapshot {
+
+    private final BatchReportReader reportReader;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "reportReader" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)